client/client.go

```diff
@@ -56,7 +56,27 @@ func (c *Client) SetTimeout(timeout time.Duration) {
 	c.HTTPClient.Timeout = timeout
 }
 
+func extractContentTypeHeader(headers map[string]interface{}) (cType string){
+		headerType, ok := headers["Content-Type"]
+			if !ok {
+				return urlEncodedContentType
+			  } 
+			return headerType.(string)
+}
+
+func decodeJsonPayload(jsonBody string)(s []byte){
+	prefixTrimmedString := jsonBody[2:]
```

Comment:
This will panic if we attempt to read a response body with less than 2 bytes (for example, "0")

Reply:
This method is removed now